store/src/java/com/zimbra/cs/service/util/VolumeConfigUtil.java

```diff
@@ -101,38 +102,133 @@ public static void validateCreateVolumeRequest(CreateVolumeRequest request, Volu
         }
 
         if (enumStoreType.equals(Volume.StoreType.EXTERNAL)) {
-            // validate use in frequent access
-            if (false != volInfoRequest.getVolumeExternalInfo().isUseInFrequentAccess() &&
-                true != volInfoRequest.getVolumeExternalInfo().isUseInFrequentAccess()) {
-                throw VolumeServiceException.INVALID_REQUEST("Volume UseInFrequentAccess can be TRUE, FALSE or OPTIONAL(don't provide)", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS);
-            }
 
-            // validate use intelligent tiering
-            if (false != volInfoRequest.getVolumeExternalInfo().isUseIntelligentTiering() &&
-                true != volInfoRequest.getVolumeExternalInfo().isUseIntelligentTiering()) {
-                throw VolumeServiceException.INVALID_REQUEST("Volume UseIntelligentTiering can be TRUE, FALSE or OPTIONAL(don't provide)", VolumeServiceException.BAD_VOLUME_USE_INTELLIGENT_TIERING);
+            // validate storage type
+            String storageTypeS3 = volInfoRequest.getVolumeExternalInfo().getStorageType();
+            String storageTypeOpenIO = volInfoRequest.getVolumeExternalOpenIOInfo().getStorageType();
+
+            if (!storageTypeS3.equalsIgnoreCase(AdminConstants.A_VOLUME_S3) && !AdminConstants.A_VOLUME_OPEN_IO.equalsIgnoreCase(storageTypeOpenIO)) {
+                throw VolumeServiceException.INVALID_REQUEST("Volume Storage Type can be only S3 or OpenIO", VolumeServiceException.BAD_VOLUME_STORAGE_TYPE);
             }
 
-            // validate storage type
-            String storageType = volInfoRequest.getVolumeExternalInfo().getStorageType();
-            if (false == storageType.equals("S3")) {
-                throw VolumeServiceException.INVALID_REQUEST("Volume Storage Type can be only S3", VolumeServiceException.BAD_VOLUME_STORAGE_TYPE);
+            if(storageTypeS3.equalsIgnoreCase(AdminConstants.A_VOLUME_S3)) {
+                // validate use in frequent access
+                if (false != volInfoRequest.getVolumeExternalInfo().isUseInFrequentAccess() &&
```

Comment:
A boolean condition would either true or false, how is it possible it can't be true and false at the same time?

Reply:
@inpati and @rupeshtelus  this is a boolean value(True/False), it is never gone be a different value incase you are trying to validate it here :(

Reply:
Removed the validation, since it is a boolean (not a Boolean), therefore, It can't have any value other than false or true